### ISSUE TITLE
crash with ssh connection

### ISSUE THREAD

**Issue by agaldemas** [haproxy/haproxy#41]
## Output of `haproxy -vv` and `uname -a`
```
haproxy_1      | Restarting OpenBSD Secure Shell server: sshd.
haproxy_1      | [NOTICE] 041/132231 (1) : New worker #1 (22) forked




haproxy_1      | [WARNING] 041/132346 (1) : Former worker #589340752 (24) exited with code 255 (Unknown signal 127)
haproxy_1      | *** Error in `haproxy': free(): invalid pointer: 0x000055a12320a048 ***
haproxy_1      | ======= Backtrace: =========
haproxy_1      | /lib/x86_64-linux-gnu/libc.so.6(+0x70bfb)[0x7f793cb43bfb]
haproxy_1      | /lib/x86_64-linux-gnu/libc.so.6(+0x76fc6)[0x7f793cb49fc6]
haproxy_1      | /lib/x86_64-linux-gnu/libc.so.6(+0x7780e)[0x7f793cb4a80e]
haproxy_1      | haproxy(+0xd51d8)[0x55a122efb1d8]
haproxy_1      | haproxy(__signal_process_queue+0x8b)[0x55a122f86c8b]
haproxy_1      | haproxy(+0xd74a7)[0x55a122efd4a7]
haproxy_1      | haproxy(main+0x1c94)[0x55a122e595d4]
haproxy_1      | /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf1)[0x7f793caf32e1]
haproxy_1      | haproxy(_start+0x2a)[0x55a122e59afa]
haproxy_1      | ======= Memory map: ========
haproxy_1      | 55a122e26000-55a123002000 r-xp 00000000 08:01 527449                     /usr/local/sbin/haproxy
haproxy_1      | 55a123201000-55a123206000 r--p 001db000 08:01 527449                     /usr/local/sbin/haproxy
haproxy_1      | 55a123206000-55a12321f000 rw-p 001e0000 08:01 527449                     /usr/local/sbin/haproxy
haproxy_1      | 55a12321f000-55a12323b000 rw-p 00000000 00:00 0 
haproxy_1      | 55a124fd3000-55a12509d000 rw-p 00000000 00:00 0                          [heap]
haproxy_1      | 7f7934000000-7f7934021000 rw-p 00000000 00:00 0 
haproxy_1      | 7f7934021000-7f7938000000 ---p 00000000 00:00 0 
haproxy_1      | 7f793bc61000-7f793bc77000 r-xp 00000000 08:01 407990                     /lib/x86_64-linux-gnu/libgcc_s.so.1
haproxy_1      | 7f793bc77000-7f793be76000 ---p 00016000 08:01 407990                     /lib/x86_64-linux-gnu/libgcc_s.so.1
haproxy_1      | 7f793be76000-7f793be77000 r--p 00015000 08:01 407990                     /lib/x86_64-linux-gnu/libgcc_s.so.1
haproxy_1      | 7f793be77000-7f793be78000 rw-p 00016000 08:01 407990                     /lib/x86_64-linux-gnu/libgcc_s.so.1
haproxy_1      | 7f793be78000-7f793be8c000 r-xp 00000000 08:01 408031                     /lib/x86_64-linux-gnu/libresolv-2.24.so
haproxy_1      | 7f793be8c000-7f793c08b000 ---p 00014000 08:01 408031                     /lib/x86_64-linux-gnu/libresolv-2.24.so
haproxy_1      | 7f793c08b000-7f793c08c000 r--p 00013000 08:01 408031                     /lib/x86_64-linux-gnu/libresolv-2.24.so
haproxy_1      | 7f793c08c000-7f793c08d000 rw-p 00014000 08:01 408031                     /lib/x86_64-linux-gnu/libresolv-2.24.so
haproxy_1      | 7f793c08d000-7f793c08f000 rw-p 00000000 00:00 0 
haproxy_1      | 7f793c08f000-7f793c094000 r-xp 00000000 08:01 408010                     /lib/x86_64-linux-gnu/libnss_dns-2.24.so
haproxy_1      | 7f793c094000-7f793c293000 ---p 00005000 08:01 408010                     /lib/x86_64-linux-gnu/libnss_dns-2.24.so
haproxy_1      | 7f793c293000-7f793c294000 r--p 00004000 08:01 408010                     /lib/x86_64-linux-gnu/libnss_dns-2.24.so
haproxy_1      | 7f793c294000-7f793c295000 rw-p 00005000 08:01 408010                     /lib/x86_64-linux-gnu/libnss_dns-2.24.so
haproxy_1      | 7f793c295000-7f793c29f000 r-xp 00000000 08:01 408012                     /lib/x86_64-linux-gnu/libnss_files-2.24.so
haproxy_1      | 7f793c29f000-7f793c49f000 ---p 0000a000 08:01 408012                     /lib/x86_64-linux-gnu/libnss_files-2.24.so
haproxy_1      | 7f793c49f000-7f793c4a0000 r--p 0000a000 08:01 408012                     /lib/x86_64-linux-gnu/libnss_files-2.24.so
haproxy_1      | 7f793c4a0000-7f793c4a1000 rw-p 0000b000 08:01 408012                     /lib/x86_64-linux-gnu/libnss_files-2.24.so
haproxy_1      | 7f793c4a1000-7f793c4a7000 rw-p 00000000 00:00 0 
haproxy_1      | 7f793c4a7000-7f793c4b2000 r-xp 00000000 08:01 408016                     /lib/x86_64-linux-gnu/libnss_nis-2.24.so
haproxy_1      | 7f793c4b2000-7f793c6b1000 ---p 0000b000 08:01 408016                     /lib/x86_64-linux-gnu/libnss_nis-2.24.so
haproxy_1      | 7f793c6b1000-7f793c6b2000 r--p 0000a000 08:01 408016                     /lib/x86_64-linux-gnu/libnss_nis-2.24.so
haproxy_1      | 7f793c6b2000-7f793c6b3000 rw-p 0000b000 08:01 408016                     /lib/x86_64-linux-gnu/libnss_nis-2.24.so
haproxy_1      | 7f793c6b3000-7f793c6c7000 r-xp 00000000 08:01 408006                     /lib/x86_64-linux-gnu/libnsl-2.24.so
haproxy_1      | 7f793c6c7000-7f793c8c7000 ---p 00014000 08:01 408006                     /lib/x86_64-linux-gnu/libnsl-2.24.so
haproxy_1      | 7f793c8c7000-7f793c8c8000 r--p 00014000 08:01 408006                     /lib/x86_64-linux-gnu/libnsl-2.24.so
haproxy_1      | 7f793c8c8000-7f793c8c9000 rw-p 00015000 08:01 408006                     /lib/x86_64-linux-gnu/libnsl-2.24.so
haproxy_1      | 7f793c8c9000-7f793c8cb000 rw-p 00000000 00:00 0 
haproxy_1      | 7f793c8cb000-7f793c8d2000 r-xp 00000000 08:01 408008                     /lib/x86_64-linux-gnu/libnss_compat-2.24.so
haproxy_1      | 7f793c8d2000-7f793cad1000 ---p 00007000 08:01 408008                     /lib/x86_64-linux-gnu/libnss_compat-2.24.so
haproxy_1      | 7f793cad1000-7f793cad2000 r--p 00006000 08:01 408008                     /lib/x86_64-linux-gnu/libnss_compat-2.24.so
haproxy_1      | 7f793cad2000-7f793cad3000 rw-p 00007000 08:01 408008                     /lib/x86_64-linux-gnu/libnss_compat-2.24.so
haproxy_1      | 7f793cad3000-7f793cc68000 r-xp 00000000 08:01 407972                     /lib/x86_64-linux-gnu/libc-2.24.so
haproxy_1      | 7f793cc68000-7f793ce68000 ---p 00195000 08:01 407972                     /lib/x86_64-linux-gnu/libc-2.24.so
haproxy_1      | 7f793ce68000-7f793ce6c000 r--p 00195000 08:01 407972                     /lib/x86_64-linux-gnu/libc-2.24.so
haproxy_1      | 7f793ce6c000-7f793ce6e000 rw-p 00199000 08:01 407972                     /lib/x86_64-linux-gnu/libc-2.24.so
haproxy_1      | 7f793ce6e000-7f793ce72000 rw-p 00000000 00:00 0 
haproxy_1      | 7f793ce72000-7f793cee4000 r-xp 00000000 08:01 408028                     /lib/x86_64-linux-gnu/libpcre.so.3.13.3
haproxy_1      | 7f793cee4000-7f793d0e3000 ---p 00072000 08:01 408028                     /lib/x86_64-linux-gnu/libpcre.so.3.13.3
haproxy_1      | 7f793d0e3000-7f793d0e4000 r--p 00071000 08:01 408028                     /lib/x86_64-linux-gnu/libpcre.so.3.13.3
haproxy_1      | 7f793d0e4000-7f793d0e5000 rw-p 00072000 08:01 408028                     /lib/x86_64-linux-gnu/libpcre.so.3.13.3
haproxy_1      | 7f793d0e5000-7f793d0e7000 r-xp 00000000 08:01 408724                     /usr/lib/x86_64-linux-gnu/libpcreposix.so.3.13.3
haproxy_1      | 7f793d0e7000-7f793d2e6000 ---p 00002000 08:01 408724                     /usr/lib/x86_64-linux-gnu/libpcreposix.so.3.13.3
haproxy_1      | 7f793d2e6000-7f793d2e7000 r--p 00001000 08:01 408724                     /usr/lib/x86_64-linux-gnu/libpcreposix.so.3.13.3
haproxy_1      | 7f793d2e7000-7f793d2e8000 rw-p 00002000 08:01 408724                     /usr/lib/x86_64-linux-gnu/libpcreposix.so.3.13.3
haproxy_1      | 7f793d2e8000-7f793d3eb000 r-xp 00000000 08:01 407997                     /lib/x86_64-linux-gnu/libm-2.24.so
haproxy_1      | 7f793d3eb000-7f793d5ea000 ---p 00103000 08:01 407997                     /lib/x86_64-linux-gnu/libm-2.24.so
haproxy_1      | 7f793d5ea000-7f793d5eb000 r--p 00102000 08:01 407997                     /lib/x86_64-linux-gnu/libm-2.24.so
haproxy_1      | 7f793d5eb000-7f793d5ec000 rw-p 00103000 08:01 407997                     /lib/x86_64-linux-gnu/libm-2.24.so
haproxy_1      | 7f793d5ec000-7f793d623000 r-xp 00000000 08:01 527434                     /usr/lib/x86_64-linux-gnu/liblua5.3.so.0.0.0
haproxy_1      | 7f793d623000-7f793d822000 ---p 00037000 08:01 527434                     /usr/lib/x86_64-linux-gnu/liblua5.3.so.0.0.0
haproxy_1      | 7f793d822000-7f793d824000 r--p 00036000 08:01 527434                     /usr/lib/x86_64-linux-gnu/liblua5.3.so.0.0.0
haproxy_1      | 7f793d824000-7f793d825000 rw-p 00038000 08:01 527434                     /usr/lib/x86_64-linux-gnu/liblua5.3.so.0.0.0
haproxy_1      | 7f793d825000-7f793da8f000 r-xp 00000000 08:01 527430                     /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1
haproxy_1      | 7f793da8f000-7f793dc8f000 ---p 0026a000 08:01 527430                     /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1
haproxy_1      | 7f793dc8f000-7f793dcad000 r--p 0026a000 08:01 527430                     /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1
haproxy_1      | 7f793dcad000-7f793dcbb000 rw-p 00288000 08:01 527430                     /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1
haproxy_1      | 7f793dcbb000-7f793dcbe000 rw-p 00000000 00:00 0 
haproxy_1      | 7f793dcbe000-7f793dd21000 r-xp 00000000 08:01 527435                     /usr/lib/x86_64-linux-gnu/libssl.so.1.1
haproxy_1      | 7f793dd21000-7f793df20000 ---p 00063000 08:01 527435                     /usr/lib/x86_64-linux-gnu/libssl.so.1.1
haproxy_1      | 7f793df20000-7f793df24000 r--p 00062000 08:01 527435                     /usr/lib/x86_64-linux-gnu/libssl.so.1.1
haproxy_1      | 7f793df24000-7f793df2a000 rw-p 00066000 08:01 527435                     /usr/lib/x86_64-linux-gnu/libssl.so.1.1
haproxy_1      | 7f793df2a000-7f793df31000 r-xp 00000000 08:01 408033                     /lib/x86_64-linux-gnu/librt-2.24.so
haproxy_1      | 7f793df31000-7f793e130000 ---p 00007000 08:01 408033                     /lib/x86_64-linux-gnu/librt-2.24.so
haproxy_1      | 7f793e130000-7f793e131000 r--p 00006000 08:01 408033                     /lib/x86_64-linux-gnu/librt-2.24.so
haproxy_1      | 7f793e131000-7f793e132000 rw-p 00007000 08:01 408033                     /lib/x86_64-linux-gnu/librt-2.24.so
haproxy_1      | 7f793e132000-7f793e14a000 r-xp 00000000 08:01 408029                     /lib/x86_64-linux-gnu/libpthread-2.24.so
haproxy_1      | 7f793e14a000-7f793e349000 ---p 00018000 08:01 408029                     /lib/x86_64-linux-gnu/libpthread-2.24.so
haproxy_1      | 7f793e349000-7f793e34a000 r--p 00017000 08:01 408029                     /lib/x86_64-linux-gnu/libpthread-2.24.so
haproxy_1      | 7f793e34a000-7f793e34b000 rw-p 00018000 08:01 408029                     /lib/x86_64-linux-gnu/libpthread-2.24.so
haproxy_1      | 7f793e34b000-7f793e34f000 rw-p 00000000 00:00 0 
haproxy_1      | 7f793e34f000-7f793e352000 r-xp 00000000 08:01 407982                     /lib/x86_64-linux-gnu/libdl-2.24.so
haproxy_1      | 7f793e352000-7f793e551000 ---p 00003000 08:01 407982                     /lib/x86_64-linux-gnu/libdl-2.24.so
haproxy_1      | 7f793e551000-7f793e552000 r--p 00002000 08:01 407982                     /lib/x86_64-linux-gnu/libdl-2.24.so
haproxy_1      | 7f793e552000-7f793e553000 rw-p 00003000 08:01 407982                     /lib/x86_64-linux-gnu/libdl-2.24.so
haproxy_1      | 7f793e553000-7f793e56c000 r-xp 00000000 08:01 408054                     /lib/x86_64-linux-gnu/libz.so.1.2.8
haproxy_1      | 7f793e56c000-7f793e76b000 ---p 00019000 08:01 408054                     /lib/x86_64-linux-gnu/libz.so.1.2.8
haproxy_1      | 7f793e76b000-7f793e76c000 r--p 00018000 08:01 408054                     /lib/x86_64-linux-gnu/libz.so.1.2.8
haproxy_1      | 7f793e76c000-7f793e76d000 rw-p 00019000 08:01 408054                     /lib/x86_64-linux-gnu/libz.so.1.2.8
haproxy_1      | 7f793e76d000-7f793e775000 r-xp 00000000 08:01 407980                     /lib/x86_64-linux-gnu/libcrypt-2.24.so
haproxy_1      | 7f793e775000-7f793e975000 ---p 00008000 08:01 407980                     /lib/x86_64-linux-gnu/libcrypt-2.24.so
haproxy_1      | 7f793e975000-7f793e976000 r--p 00008000 08:01 407980                     /lib/x86_64-linux-gnu/libcrypt-2.24.so
haproxy_1      | 7f793e976000-7f793e977000 rw-p 00009000 08:01 407980                     /lib/x86_64-linux-gnu/libcrypt-2.24.so
haproxy_1      | 7f793e977000-7f793e9a5000 rw-p 00000000 00:00 0 
haproxy_1      | 7f793e9a5000-7f793e9c8000 r-xp 00000000 08:01 407954                     /lib/x86_64-linux-gnu/ld-2.24.so
haproxy_1      | 7f793eb5a000-7f793ebc2000 rw-p 00000000 00:00 0 
haproxy_1      | 7f793ebc4000-7f793ebc8000 rw-p 00000000 00:00 0 
haproxy_1      | 7f793ebc8000-7f793ebc9000 r--p 00023000 08:01 407954                     /lib/x86_64-linux-gnu/ld-2.24.so
haproxy_1      | 7f793ebc9000-7f793ebca000 rw-p 00024000 08:01 407954                     /lib/x86_64-linux-gnu/ld-2.24.so
haproxy_1      | 7f793ebca000-7f793ebcb000 rw-p 00000000 00:00 0 
haproxy_1      | 7ffd89cfa000-7ffd89d1b000 rw-p 00000000 00:00 0                          [stack]
haproxy_1      | 7ffd89d2e000-7ffd89d30000 r--p 00000000 00:00 0                          [vvar]
haproxy_1      | 7ffd89d30000-7ffd89d32000 r-xp 00000000 00:00 0                          [vdso]
haproxy_1      | ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]

```

uname - a:
```
root@a0ba2b3cb947:/# uname -a
Linux a0ba2b3cb947 4.9.125-linuxkit #1 SMP Fri Sep 7 08:20:28 UTC 2018 x86_64 GNU/Linux

```

## What's the configuration?
run from an image (name: haproxy-spec) with Dockerfile:
```
FROM haproxy:1.9

RUN apt-get update && apt-get install -y openssh-server socat
RUN mkdir /var/run/sshd
RUN echo 'root:screencast' | chpasswd
RUN sed -i 's/PermitRootLogin prohibit-password/PermitRootLogin yes/' /etc/ssh/sshd_config

# SSH login fix. Otherwise user is kicked off after login
RUN sed 's@session\s*required\s*pam_loginuid.so@session optional pam_loginuid.so@g' -i /etc/pam.d/sshd

ENV NOTVISIBLE "in users profile"
RUN echo "export VISIBLE=now" >> /etc/profile

#
COPY docker-entrypoint.sh /
RUN chmod +x /docker-entrypoint.sh

```
the modified docker-entrypoint.sh (added a line with `service ssh restart`):
```
#!/bin/sh
service ssh restart
set -e

# first arg is `-f` or `--some-option`
if [ "${1#-}" != "$1" ]; then
        set -- haproxy "$@"
fi

if [ "$1" = 'haproxy' ]; then
        shift # "haproxy"
        # if the user wants "haproxy", let's add a couple useful flags
        #   -W  -- "master-worker mode" (similar to the old "haproxy-systemd-wrapper"; allows for reload via "SIGUSR2")
        #   -db -- disables background mode
        set -- haproxy -W -db "$@"
fi

exec "$@"

```

config in docker-compose.yml:
```
  haproxy:
    image: haproxy-spec
    # image: dockercloud/haproxy
    restart: always
    ports:
      - "80:80"
      - "2222:22"
    volumes:
      #- /var/run/docker.sock:/var/run/docker.sock
      - ./haproxy/haproxy.cfg:/usr/local/etc/haproxy/haproxy.cfg


```

haproxy configuration 
```
global
    #log /dev/log local0
    #log /dev/log local1 notice
    #chroot /var/lib/haproxy
    #stats socket /run/haproxy/admin.sock mode 660 level admin
    #stats timeout 30s
    user root
    group root
    daemon

    # Default SSL material locations
    ca-base /etc/ssl/certs
    crt-base /etc/ssl/private

    # Default ciphers to use on SSL-enabled listening sockets.
    # For more information, see ciphers(1SSL). This list is from:
    #  https://hynek.me/articles/hardening-your-web-servers-ssl-ciphers/
    ssl-default-bind-ciphers ECDH+AESGCM:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:ECDH+3DES:DH+3DES:RSA+AESGCM:RSA+AES:RSA+3DES:!aNULL:!MD5:!DSS
    ssl-default-bind-options no-sslv3

defaults
    log global
    mode http
    #option httplog
    #option dontlognull
    timeout connect 5000
    timeout client  50000
    timeout server  50000

# frontend ssh-in
#     bind *:22
#     mode tcp
#     option tcplog
#     default_backend back_ssh
#
# backend back_ssh
#     mode tcp
#     option tcp-check
#     tcp-check expect string SSH-2.0-
#     server ssh_server haproxy:22 check

frontend http-in
    bind *:80
    default_backend servers

backend servers
        #server 'orion-eridanis.com' orion:1026 maxconn 32
        server 'idm-eridanis.com' keyrock:3000 maxconn 32

```

## Steps to reproduce the behavior

1. try to connect via ssh to the container: ssh root@localhost:2222
2. ask for password, connection refused each time, even with the right password, (3-4 times retry)
3. at the end haproxy crashes with V1.9.4, no connection possible, 
Note: with 1.8 (1.8.18), no haproxy crash but still can't connect via ssh ? 

## Expected behavior:
1: no crash
2: to be able to connect through ssh (may be the config is not OK ?)

**Comment by TimWolla** [comment]
Something does not add up here: With the configuration you posted HAProxy is not even involved in the SSH connection, because the relevant frontend is commented out. Commenting it in is not possible, because OpenSSH is already listening on port 22, thus HAProxy can't bind to 22.

Can you create a self-contained Dockerfile that does not rely on external services (such as `keyrock`)?

**Comment by agaldemas** [comment]
Hello,
I reproduced the issue after removing any external service, even any container except portainer, 

- with version 1.8 as root image, there's no crash, but seems that the thread die, 
- with 1.9 got the invalid pointer error reported. 


I discover this issue while trying to use an [haproxy-wi](https://github.com/Aidaho12/haproxy-wi) container to connect to an haproxy container (through ssh).

I started with 1.8 from dockerhub image, as I didn't manage to connect to the container through ssh, 
I tried version 1.9 and haproxy crashed each time I tried to connect via ssh to the container (through port 22)

May be there is a particularity in my docker configuration (under MacOS), I will try on a virtualbox VM
May be my post-installation in Dockerfile break something ?
Or service ssh restart before starting haproxy in docker-entrypoint.sh ?

**Comment by TimWolla** [comment]
>  I reproduced the issue after removing any external service, even any container except portainer,

Can you provide that self-contained `Dockerfile` with a list of exact steps, please? Ideally I should be able to just `docker build` the image, `docker run` the container and then connect to reproduce the issue.

**Comment by agaldemas** [comment]
Here is the Dockerfile(.txt)
[Dockerfile.txt](https://github.com/haproxy/haproxy/files/2862151/Dockerfile.txt)
just change the version of the base haproxy image, 1.8 or 1.9

But, think it can be considered as a kind of corner case...
Because as an haproxy newbie, but as experimented senior SW dev&architect, I din't read a lot before starting testing,
and later I realized it's not a useful idea to install openssh on the same container, instead of having a specific one for it, beside haproxy, which is a smarter and better practice ;O).

Anyway sometimes strange corner case bugs, can reveal a true hidden issue, 

You have the docker file, the docker-compose file part above,
to reproduce the issue, try to ssh the haproxy with ssh root@localhost:2222, or even form the haproxy host itself
answer 'screencast' the password (or whatever, it doesn't matter), at the end of process on my configuration the container die in version 1.9 with invalid pointer exception reported.
Waitng for your feedback, Have a nice test

**Comment by TimWolla** [comment]
> Waitng for your feedback, Have a nice test

What is the HAProxy configuration? As outlined in my initial comment I am unable to reproduce the issue with the configuration you provided above, because it relies on external services and does not start because it is unable to resolve then.

**Comment by agaldemas** [comment]
here is the haproxy.conf:

```
global
    #log /dev/log local0
    #log /dev/log local1 notice
    #chroot /var/lib/haproxy
    #stats socket /run/haproxy/admin.sock mode 660 level admin
    #stats timeout 30s
    user root
    group root
    daemon

    # Default SSL material locations
    ca-base /etc/ssl/certs
    crt-base /etc/ssl/private

    # Default ciphers to use on SSL-enabled listening sockets.
    # For more information, see ciphers(1SSL). This list is from:
    #  https://hynek.me/articles/hardening-your-web-servers-ssl-ciphers/
    ssl-default-bind-ciphers ECDH+AESGCM:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:ECDH+3DES:DH+3DES:RSA+AESGCM:RSA+AES:RSA+3DES:!aNULL:!MD5:!DSS
    ssl-default-bind-options no-sslv3

defaults
    log global
    mode http
    #option httplog
    #option dontlognull
    timeout connect 5000
    timeout client  50000
    timeout server  50000
```
any other block is commented.
I you don't reproduce it, probably it's an issue with my configuration (Docker under MacOS),
On which environment are you trying to reproduce ?

**Comment by TimWolla** [comment]
Okay. HAProxy clearly is not involved within the connection here (it does not even listen to any ports at all) and indeed I'm unable to reproduce the issue with the `haproxy:1.9` image using Docker 18.09.2 on Ubuntu.

The only thing that might be able to give some insight is the core dump or attaching to the running process with gdb and waiting for the crash to get a backtrace.

**Comment by agaldemas** [comment]
I know that's why it is so weird crash !
It has been a long time I used gdb, never with Docker
Can you give me hints for that
Then I could generate the backtrace and send it to you
CU

Le jeu. 14 févr. 2019 à 11:57, Tim Düsterhus <notifications@github.com> a
écrit :

**Comment by TimWolla** [comment]
I guess the easiest way is finding the `PID` of the worker process and then attaching to it (it's the one with the higher PID). In my example I killed the worker with `kill -SEGV` for a demonstration. I'm not sure whether you can do this easily on your Mac, because Docker runs in lightweight VM if I understand it correctly.

```
[timwolla@/t/crasher]ps aux |grep haproxy
root     19648  0.0  0.0   1048     4 ?        Ss   13:29   0:00 /dev/init -- /docker-entrypoint.sh haproxy -f /usr/local/etc/haproxy/haproxy.cfg
root     19740  0.0  0.0  43548  4428 ?        S    13:29   0:00 haproxy -W -db -f /usr/local/etc/haproxy/haproxy.cfg
root     19793  0.0  0.0  43548  2756 ?        S    13:29   0:00 haproxy -W -db -f /usr/local/etc/haproxy/haproxy.cfg
timwolla 19925  0.0  0.0  15776   968 pts/19   S+   13:30   0:00 grep --color=auto haproxy
[timwolla@/t/crasher]sudo gdb -p 19793
*snip*

warning: Unable to find libthread_db matching inferior's thread library, thread debugging will not be available.
0x00007f4ed48e70a3 in epoll_wait () from target:/lib/x86_64-linux-gnu/libc.so.6
(gdb) cont
Continuing.

Program received signal SIGSEGV, Segmentation fault.
0x00007f4ed48e70a3 in epoll_wait () from target:/lib/x86_64-linux-gnu/libc.so.6
(gdb) bt full
#0  0x00007f4ed48e70a3 in epoll_wait () from target:/lib/x86_64-linux-gnu/libc.so.6
No symbol table info available.
#1  0x000055b3bd5c677d in _do_poll (p=<optimized out>, exp=<optimized out>) at src/ev_epoll.c:156
        timeout = 1000
        status = <optimized out>
        fd = <optimized out>
        count = <optimized out>
        updt_idx = <optimized out>
        old_fd = <optimized out>
#2  0x000055b3bd6681c2 in run_poll_loop () at src/haproxy.c:2652
        next = <optimized out>
        exp = <optimized out>
#3  run_thread_poll_loop (data=<optimized out>) at src/haproxy.c:2684
        ptif = <optimized out>
        ptdf = <optimized out>
        start_lock = 0
#4  0x000055b3bd5c4283 in main (argc=<optimized out>, argv=<optimized out>) at src/haproxy.c:3313
        tids = 0x55b3bddbda90
        threads = 0x55b3bddbdab0
        i = <optimized out>
        old_sig = {__val = {0, 139976553102080, 24, 139976570649344, 140733718668488, 140733718668440, 5, 139976549830500, 140733718668488, 94230472160944, 94230472237504, 139976564790982, 335544638, 
            367767529555655936, 5, 139976570649360}}
        blocked_sig = {__val = {18446744067199990583, 18446744073709551615 <repeats 15 times>}}
        err = <optimized out>
        retry = <optimized out>
        limit = {rlim_cur = 4030, rlim_max = 4030}
        errmsg = '\000' <repeats 24 times>, "\t\037q\325N\177\000\000 \"O\037\377\177\000\000\005\000\000\000N\177\000\000A\000\000\000\000\000\000\000p\000\000\000\000\000\000\000\001\000\000\000\000\000\000\000\005\000\000\000\061", '\000' <repeats 19 times>, "[\000\000"
        pidfd = <optimized out>
```

**Comment by TimWolla** [comment]
This appears to be a duplicate of #29. /cc @lukastribus

**Comment by capflam** [comment]
Old enough without activity to be closed now.
